city.lua

```diff
@@ -1390,25 +1405,130 @@ local function findUpgradableCells(city, limit, upgradeTo)
     return upgradeCells
 end
 
+--- NOTE: this function DOES NOT clear city grid cell, must be done manually if needed
+--- @param city City | nil
+--- @param unit_number number
+local function removeBuilding(city, unit_number)
+    log("removeBuilding(): unit_number: ".. serpent.line(unit_number))
+
+    -- remove anything related in reverse order
+    removeHouseLight(unit_number)
+
+    -- this also checks (unit_number == nil)
+    local building = Util.getGlobalBuilding(unit_number)
+    if building == nil then
+        return
+    end
+    --log("removeBuilding(): building: ".. serpent.line(building))
+
+    --- NOTE: switch city reference block, common to all the checks below
+    city = city or Util.findCityById(building.cityId)
+    if city == nil then
+        log("removeBuilding(): ERROR: unable to get city reference!")
+    end
+    -- WARN: this weird case should never happen, but we must always check
+    if city ~= nil and city.id ~= building.cityId then
+        local msg = string.format("removeBuilding(): ERROR: city.id: %s doesn't match building.cityId: %s, unit_number: %d",
```

Comment:
Let's prefix this with "Tycoon" so players know which mod this message is coming from.